### DISCUSSION TITLE
HammerDb doesn't like database names with dots or only digits.

### DISCUSSION THREAD

**Discussion by i25652** [TPC-Council/HammerDB#85] [Q&A] [answered]
Hi Steve,  

Here is another gem into collection of rare bugs.
HammerDb doesn't like database names consisting only of digits or with special chars like my.database. As expected the error comes from SQL Server and it says invalid syntax.
I understand this and wrapped $db into [ ]  in mssqlserver\mssqlsoltp.tcl in sql statements such as create database or use or alter.
Now I get presumably TCL error saying about invalid command.
Please check.

**Comment by sm-shaw** [comment, marked as answer]
This is the same as the with the Server name which is handled differently: 
As a workaround you can wrap the name in curly braces as follows: 
`{[1234.5678]}`
This will work in the build script without any further work. For the driver script you need to edit the driver script that is loaded and remove the double quotes, so change this:
`set database "{[1234.5678]}";# Database containing the TPC Schema`
to this:
`set database {[1234.5678]};# Database containing the TPC Schema`
This is essentially what the Server name already does so HammerDB wraps it in curly braces in the background and removes the quotes in the driver script. 
Is this something that would be considered needed for the database name as well or is the workaround above sufficient?

**Comment by sm-shaw** [comment]
No  further response - making assumption workaround given is sufficient for case when special characters are used in the database name.
